### PR TITLE
Turn off Overprovisioning feature (#490)

### PR DESCRIPTION
**Pull request by jinhyunr**

### CODE REVIEW

**Comment by jinhyunr** [comment]
config flag added

File: config/waagent.conf [hunk 1]
```diff
@@ -90,3 +90,8 @@ OS.OpensslPath=None
 
 # Determine the update family, this should not be changed
 # AutoUpdate.GAFamily=Prod
+
+# Determine if the overprovisioning feature is enabled. If yes, block extension
+# handling until inVMArtifactsProfile.OnHold is false.
+# Default is disabled
+OverProvisioning.EnableExtensionHandlingBlock=n
```
Comment (hglkrijger) [inline config/waagent.conf]: Maybe it's just me, but this seems confusing. "Enabling a block" seems unnecessary when we could simply "disable overprovisioning".

File: azurelinuxagent/common/conf.py [hunk 2]
```diff
@@ -218,3 +218,6 @@ def get_autoupdate_enabled(conf=__conf__):
 
 def get_autoupdate_frequency(conf=__conf__):
     return conf.get_int("Autoupdate.Frequency", 3600)
+
+def get_extension_handling_block_enabled(conf=__conf__):
+    return conf.get_switch("OverProvisioning.EnableExtensionHandlingBlock", False)
```
Comment (hglkrijger) [inline azurelinuxagent/common/conf.py]: Might want to add a newline for sanity

File: azurelinuxagent/ga/exthandlers.py [hunk 3]
```diff
@@ -196,8 +196,11 @@ def handle_ext_handlers(self, etag=None):
             return
 
         in_vm_artifacts_profile = self.protocol.get_in_vm_artifacts_profile()
-        if in_vm_artifacts_profile and \
-           in_vm_artifacts_profile.is_extension_handlers_handling_on_hold():
+        # Turning off overprovisioning feature until the bug in HostGAPlugin gets fixed
```
Comment (hglkrijger) [inline azurelinuxagent/ga/exthandlers.py]: I think you can remove this comment now

File: tests/ga/test_extension.py [hunk 4]
```diff
@@ -286,23 +286,36 @@ def test_ext_handler_io_error(self, mock_fileutil, *args):
         mock_fileutil.write_file.return_value = IOError("Mock IO Error")
         exthandlers_handler.run()
 
+    # Comment out until HostGAPlugin bug is fixed and we enabled overprovisioning feature in WALA
```
Comment (hglkrijger) [inline tests/ga/test_extension.py]: you can remove this comment

File: azurelinuxagent/ga/exthandlers.py (hunk 3 above)
Comment (jinhyunr) [inline azurelinuxagent/ga/exthandlers.py]: fixed

**Pull request merged**